### PR TITLE
Feature/profile info screen test

### PR DESCRIPTION
**Pull request by aferrante1**
### **Description:**  
This pull request introduces unit tests for the `ProfileInfoScreen` to validate the display of profile details, error handling, and navigation functionality.

### **Key Changes:**  
- Added tests to verify the display of profile details (name, academic level, section, and price) when the profile is available.
- Implemented a test to ensure the "Error loading profile..." message is displayed when the profile data is `null`.
- Created a test to check that clicking the close button triggers the navigation back action.
- Updated the mocking for `ListProfilesViewModel` and `LessonViewModel` to simulate various states, including valid profile, null profile, and empty lessons list.

### **Motivation:**  
These tests are designed to enhance the reliability of the `ProfileInfoScreen` by ensuring the UI behaves correctly in different scenarios. Testing various edge cases improves the app's robustness and user experience.

### **Testing:**  
- Ran the newly added tests to confirm that all cases pass, including valid profile data and null profile scenarios.
- Verified that the tests accurately detect issues by intentionally failing one to see the expected failure behavior.
- Checked that the navigation test correctly triggers the back navigation action when the close button is clicked.

### **Affected Files**
ProfileInfoScreenTest.kt: Added new unit tests for ProfileInfoScreen to validate different scenarios (profile details display, error handling, and navigation).

### CODE REVIEW

**Review by LeaBlancher — COMMENTED** [review]
This is very great and it test a great part of the screen. Maybe two suggestions. 
- First try to clean up a little bit the code because they is a lot of lines of code that are commented and there are also no used variables that can be removed. 
- A second point could be to kind of dupplicate you test profileInfoScreenDisplaysProfileDetails_whenProfileIsNotNull() to test it using a student Profile (in addition to the your actual test that does it for the Tutor Profile)
If you need help or if you want I can do these updates without probelm!

File: app/src/androidTest/java/com/github/se/project/ui/profile/ProfileInfoScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          academicLevel = AcademicLevel.MA2,
+          schedule = List(7) { List(12) { 0 } },
+          price = 50)
+  private val mockProfileFlow = MutableStateFlow<Profile?>(mockProfile)
+
+  val mockLesson =
+      listOf(
+          Lesson("1", "Math Tutoring", "2", Subject.ALGEBRA, listOf(Language.FRENCH), "12345"),
+          Lesson("2", "Physics Tutoring", "3", Subject.PHYSICS, listOf(Language.FRENCH), "12345"))
+  private val mockLessonFlow = MutableStateFlow<List<Lesson>>(mockLesson)
```
[the remark this answers is not in the document]
Comment (aferrante1) [reply app/src/androidTest/java/com/github/se/project/ui/profile/ProfileInfoScreenTest.kt:59]: Yes I agree, will do!

File: app/src/androidTest/java/com/github/se/project/ui/profile/ProfileInfoScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+          navigationActions = mockNavigationActions,
+          listProfilesViewModel = mockListProfilesViewModel,
+          lessonViewModel = mockLessonViewModel)
+    }
+
+    // Check if profile details are displayed
+    composeTestRule
+        .onNodeWithTag("profileName")
+        .assertIsDisplayed()
+        .assertTextEquals("${mockTutorProfile.firstName} ${mockTutorProfile.lastName}")
```
Comment (aferrante1) [inline app/src/androidTest/java/com/github/se/project/ui/profile/ProfileInfoScreenTest.kt:124]: I did not know it was possible to write with variables inside the assertTextEquals, it's good to know thanks!

**Review by MPepito — APPROVED** [review]
Great Job @LeaBlancher for fixing the issue with these tests and that PR until late in the Night. Next sprint, we will have to make sure we better anticipate and those assigned to that task should deliver the work ahead of schedule to avoid such situation for you, sorry for that.
